File: todo-backend/src/main/java/com/todolist/service/TaskService.java
package com.todolist.service;

import com.todolist.entity.Task;
import com.todolist.entity.Topic;
import com.todolist.repository.TaskRepository;
import com.todolist.repository.TopicRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

/**
 * 任务服务类
 * 
 * 负责处理任务相关的业务逻辑，包括任务的增删改查操作
 * 任务是待办事项的基本单位，每个任务属于一个主题
 * 
 * @author todolist
 * @version 1.0
 * @since 2025-09-12
 */
@Service
public class TaskService {
    
    @Autowired
    private TaskRepository taskRepository;
    
    @Autowired
    private TopicRepository topicRepository;
    
    /**
     * 根据主题ID获取该主题下的所有任务
     * 
     * @param topicId 主题ID
     * @return List<Task> 该主题下的任务列表
     */
    public List<Task> getTasksByTopicId(Long topicId) {
        return taskRepository.findByTopicId(topicId);
    }
    
    /**
     * 根据ID获取特定任务
     * 
     * @param id 任务ID
     * @return Optional<Task> 任务信息，如果不存在则返回空
     */
    public Optional<Task> getTaskById(Long id) {
        return taskRepository.findById(id);
    }
    
    /**
     * 在指定主题下创建新任务
     * 
     * @param task 要创建的任务信息
     * @param topicId 主题ID
     * @return Task 创建成功的任务信息（包含生成的ID）
     * @throws RuntimeException 如果主题不存在则抛出异常
     */
    public Task createTask(Task task, Long topicId) {
        Topic topic = topicRepository.findById(topicId)
            .orElseThrow(() -> new RuntimeException("主题不存在，ID: " + topicId));
        task.setTopic(topic);
        return taskRepository.save(task);
    }
    
    /**
     * 更新任务信息
     * 
     * @param task 要更新的任务信息
     * @return Task 更新后的任务信息
     */
    public Task updateTask(Task task) {
        return taskRepository.save(task);
    }
    
    /**
     * 删除任务
     * 
     * @param id 要删除的任务ID
     * @return boolean 删除是否成功
     */
    public boolean deleteTask(Long id) {
        if (taskRepository.existsById(id)) {
            taskRepository.deleteById(id);
            return true;
        }
        return false;
    }
}

File: src/services/taskService.js
// src/services/taskService.js
import api from './api.js'

export const taskService = {
  // 获取主题下的所有任务
  async getTasksByTopicId(topicId) {
    try {
      console.log('获取任务列表:', topicId)
      const response = await api.get(`/tasks/topic/${topicId}`)
      console.log('任务列表响应:', response)
      return response
    } catch (error) {
      console.error('获取任务列表失败:', error)
      throw new Error(error.response?.data?.message || '获取任务列表失败')
    }
  },

  // 根据ID获取任务
  async getTaskById(id) {
    try {
      console.log('获取任务详情:', id)
      const response = await api.get(`/tasks/${id}`)
      console.log('任务详情响应:', response)
      return response
    } catch (error) {
      console.error('获取任务详情失败:', error)
      throw new Error(error.response?.data?.message || '获取任务详情失败')
    }
  },

  // 创建任务
  async createTask(taskData, topicId) {
    try {
      console.log('创建任务:', { taskData, topicId })
      const response = await api.post(`/tasks/topic/${topicId}`, taskData)
      console.log('创建任务响应:', response)
      return response
    } catch (error) {
      console.error('创建任务失败:', error)
      throw new Error(error.response?.data?.message || '创建任务失败')
    }
  },

  // 更新任务
  async updateTask(id, taskData) {
    try {
      console.log('更新任务:', { id, taskData })
      const response = await api.put(`/tasks/${id}`, taskData)
      console.log('更新任务响应:', response)
      return response
    } catch (error) {
      console.error('更新任务失败:', error)
      throw new Error(error.response?.data?.message || '更新任务失败')
    }
  },

  // 删除任务
  async deleteTask(id) {
    try {
      console.log('删除任务:', id)
      const response = await api.delete(`/tasks/${id}`)
      console.log('删除任务响应:', response)
      return response
    } catch (error) {
      console.error('删除任务失败:', error)
      throw new Error(error.response?.data?.message || '删除任务失败')
    }
  }
}
